libse/Forms/DurationsBridgeGaps.cs

```diff
@@ -5,48 +5,47 @@ namespace Nikse.SubtitleEdit.Core.Forms
+                SeLogger.Error(new InvalidOperationException(message), message);
+                return 0;
+            }
+
+            int count = subtitle.Paragraphs.Count - 1;
+            for (int i = 0; i < count; i++)
+            {
+                Paragraph cur = subtitle.Paragraphs[i];
+                Paragraph next = subtitle.Paragraphs[i + 1];
+
+                double currentGaps = next.StartTime.TotalMilliseconds - cur.EndTime.TotalMilliseconds;
+
+                // there shouldn't be adjustment if current gaps is shorter than minimum gaps or greater than maximum gaps
+                if (currentGaps < minMsBetweenLines || currentGaps > maxMs)
                 {
-                    before = $"{(next.StartTime.TotalMilliseconds - cur.EndTime.TotalMilliseconds) / TimeCode.BaseUnit:0.000}";
-                    if (divideEven && next.StartTime.TotalMilliseconds > cur.EndTime.TotalMilliseconds)
-                    {
-                        double half = (next.StartTime.TotalMilliseconds - cur.EndTime.TotalMilliseconds) / 2.0;
-                        next.StartTime.TotalMilliseconds -= half;
-                    }
-                    cur.EndTime.TotalMilliseconds = next.StartTime.TotalMilliseconds - minMsBetweenLines;
-                    if (fixedIndexes != null)
-                    {
-                        fixedIndexes.Add(i);
-                        fixedIndexes.Add(i + 1);
-                    }
-                    fixedCount++;
+                    continue;
                 }
-                var msToNext = next.StartTime.TotalMilliseconds - cur.EndTime.TotalMilliseconds;
-                if (msToNext < 2000)
+
+                // next paragraph start-time will be pull to try to meet the current parragraph
+                if (divideEven)
                 {
-                    string info;
-                    if (!string.IsNullOrEmpty(before))
-                    {
-                        info = $"{before} => {msToNext / TimeCode.BaseUnit:0.000}";
-                    }
-                    else
-                    {
-                        info = $"{msToNext / TimeCode.BaseUnit:0.000}";
-                    }
-
-                    dic?.Add(cur.ID, info);
+                    next.StartTime.TotalMilliseconds -= currentGaps / 2.0;
                 }
+
+                cur.EndTime.TotalMilliseconds = next.StartTime.TotalMilliseconds - minMsBetweenLines;
+                if (fixedIndexes != null)
+                {
+                    fixedIndexes.Add(i);
+                    fixedIndexes.Add(i + 1);
+                }
+
+                double newGaps = next.StartTime.TotalMilliseconds - cur.EndTime.TotalMilliseconds;
+                dic?.Add(cur.ID, $"{currentGaps / TimeCode.BaseUnit:0.000} => {newGaps / TimeCode.BaseUnit:0.000}");
             }
-            return fixedCount;
+
+            return fixedIndexes.Count / 2;
```

Comment:
No need for variable  `fixedCount`, total adjustment can be obtained from `fixedIndex.Count / 2` or `dic.Count`